source/code/plugins/auditlog_lib.rb

```diff
@@ -0,0 +1,161 @@
+module AuditLogModule
+  class LoggingBase
+    def log_error(text)
+    end
+  end
+
+  class RuntimeError < LoggingBase
+    def log_error(text)
+      $log.error "RuntimeError: #{text}"
+    end
+  end
+
+  class AuditLogParser
+    require 'date'
+    require 'etc'
+    require 'digest'
+    require_relative 'oms_common'
+
+    def initialize(error_handler)
+      @error_handler = error_handler
+      @sha256 = Digest::SHA256.new
+    end
+
+    SKETCH_REGEX = /type=(?<record>[A-Z_]+) msg=audit\((?<audit_id>(?<time_epoch>[0-9\.]+):(?<serial_number>[0-9]+))\): (?<additional_fields>.+)/
+    ADDITIONAL_MSG_REGEX = /(?<additional_msg>^[^=]+) [a-z0-1_\-]+=/
+    FIELDS_REGEX = /(msg=')?([a-z0-9_\-]+)=("[^"]*"|'[^']*'|[^ ]*)?/
+
+    def parse(line)
+      data = {}
+      time = Time.now.to_f
+
+      begin
+        SKETCH_REGEX.match(line) { |match|
+          data['Computer'] = OMS::Common.get_hostname
+          data['RecordType'] = match['record']
+          data['AuditID'] = match['audit_id']
+          time = match['time_epoch'].to_f
+          data['Timestamp'] = OMS::Common.format_time(time)
+          data['SerialNumber'] = match['serial_number']
+          additional_fields = match['additional_fields']
+        
+          # process the message before the first field
+          ADDITIONAL_MSG_REGEX.match(additional_fields) { |m|
+            data['AdditionalMessage'] = m['additional_msg']
+          }
+        
+          additional_fields.scan(FIELDS_REGEX) { |m|
+            name = m[1]
+            value = remove_quotations(m[2])
+            data[name] = get_typed_value(name, value)
+            friendly_name, readable_value = translate(name, value)
+            data[friendly_name] = readable_value if !readable_value.nil?
+          }
+          data['Hash'] = @sha256.hexdigest line
+        }
+      rescue => e
+        @error_handler.log_error("Unable to parse the line #{e}")
+      end
+
+      return time, data
+    end
+    
+
+    private
+
+    def remove_quotations(value)

```

Comment:
If you want to remove all quotations consider using `String.tr` otherwise rename to strip_quotations.